src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -520,12 +520,21 @@ protected override Match<SyntaxNode> ComputeBodyMatch(SyntaxNode oldBody, Syntax
 
                 // We use the parent node as a root:
                 // - for field/property initializers the root is EqualsValueClause. 
-                // - for expression-bodies the root is ArrowExpressionClauseSyntax. 
+                // - for member expression-bodies the root is ArrowExpressionClauseSyntax.
                 // - for block bodies the root is a method/operator/accessor declaration (only happens when matching expression body with a block body)
                 // - for lambdas the root is a LambdaExpression.
                 // - for query lambdas the root is the query clause containing the lambda (e.g. where).
+                // - for local functions the root is LocalFunctionStatement.
 
-                return new StatementSyntaxComparer(oldBody, newBody).ComputeMatch(oldBody.Parent, newBody.Parent, knownMatches);
+                SyntaxNode GetMatchingRoot(SyntaxNode body)
+                {
+                    var root = body.Parent;
```

Comment:
>root [](start = 24, length = 4)

Nit: perhaps rename to "parent", since at this point it might not be the root we are looking for.

Reply:
Thank you!

---
In reply to: [145001822](https://github.com/dotnet/roslyn/pull/22624#discussion_r145001822) [](ancestors = 145001822)